content/300-guides/200-deployment/110-deployment-guides/450-caveats-when-deploying-to-aws-platforms.mdx

```diff
@@ -0,0 +1,98 @@
+---
+title: 'Caveats when deploying to AWS platforms'
+metaTitle: 'Caveats when deploying to AWS platforms'
+metaDescription: 'Known caveats when deploying to an AWS platform'
+---
+
+<TopBlock>
+
+The following describes some caveats you might face when deploying to different AWS platforms.
+
+</TopBlock>
+
+## AWS Lambda + Serverless Framework + Next.js
+
+The [Serverless Framework](https://www.serverless.com/framework/docs/) is an open source CLI that enables you to manage and deploy serverless applications.
+
+The Serverless Framework can be used to deploy an [AWS Lambda function](https://www.serverless.com/framework/docs/providers/aws/guide/intro/) through Next.Js, you may however encounter issues with the Prisma binary not being copied over into the lambda function.
+
+Another known issue is the Prisma query engine losing its executable flag. This is due to the `CopyWebpackPlugin` and the `aws-lambda` component used internally by the `serverless-next.js` package.
+
+The solution is to use `WebpackPermissionsPlugin` and a community created [patched package](https://www.npmjs.com/package/serverless-nextjs-lambda-binary-fix) for the `serverless-next.js` package.
+
+A working example of a next.config.js file and serverless.yml:
+
+```js file=next.config.js
+const CopyWebpackPlugin = require('copy-webpack-plugin')
+const WebpackPermissionsPlugin = require('webpack-permissions-plugin')
+const path = require('path')
+
+module.exports = {
+  target: 'serverless',
+
+  webpack: (config, {}) => {
+    config.plugins.push(
+      new CopyWebpackPlugin([
+        {
+          from: path.join(
+            __dirname,
+            'node_modules/@prisma/client/runtime/query-engine-rhel-openssl-1.0.x'
+          ),
+          to: path.join(__dirname, '.next/serverless/pages/api/query-engine-rhel-openssl-1.0.x'),
+        },
+        {
+          from: path.join(__dirname, 'prisma/schema.prisma'),
+          to: path.join(__dirname, '.next/serverless/pages/api/'),
+        },
+      ])
+    )
+    config.plugins.push(
+      new WebpackPermissionsPlugin({
+        buildFiles: [
+          {
+            path: path.join(
+              __dirname,
+              '.next/serverless/pages/api/query-engine-rhel-openssl-1.0.x'
+            ),
+            fileMode: '755',
+          },
+        ],
+      })
+    )
+
+    // https://github.com/webpack/webpack/issues/1599
+    config.node.__dirname = false
+
+    return config
+  },
+}
+```
+
+```yml file=serverless.yml
+example:
+  component: serverless-nextjs-lambda-binary-fix@1.11.4
+```
+
+> **Note**: An SQLite database still wont be read. For this type of deployment it is suggested to us another SQL database.
```

Comment:
Why is that?

Reply:
I deleted this whole section (`AWS Lambda + Serverless Framework + Next.js`) as it was based off of a github issue that was potentially out of date. 

Its a real rabbit hole with about 20 different solutions to problems that are all related to the deployment of prisma through serverless framework to aws lambda. There doesnt appear to be one fix to recommend to users. From reading the issues it looks like the general fix is to generate the the prisam binary after deployment with a postinstall script, or via webpack somehow.